connectors/connectors-common-library/src/main/java/io/camunda/connector/common/model/aws/AWSRequest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.common.model.aws;
+
+public abstract class AWSRequest {
+  private transient String type;
+
+  public abstract void authenticate();
+
+  public abstract void configure();
+
+  public abstract Object invoke() throws InterruptedException;
```

Comment:
Authenticate and configure are natural part of invoke. Essentially, invoke depends on both. I suggest you to rebuild to remove authenticate and configure from this interface and leave just to methods `invoke()` and `type()`.